mautic/mautic
# Replace queue implementation with symfony's

| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (5.x)      | [X]
| Deprecations?                          | [X]
| BC breaks? (use the c.x branch)        | [X]
| Automated tests included?              | [X]
| User documentation | https://github.com/mautic/user-documentation/pull/195
| Related developer documentation PR URL | mautic/developer-documentation#... 

#### Description:

Replaces QueueBundle and declares new Hit handler's to process it as messages.

#### Steps to test this PR:
# How to test mautic hit messenger

1. get this PR 
2. create a landing page and an email with tracking pixel
3. open the landing page in anonymous window - now hit should be recorded
4. check the landing page's stats, the hit is recorded

## Verify Page Hit
> Make sure you have an AMQP server available, You may use doctrine as well currently I have
1. configure the messenger to use AMQP server as explained in [README.md](https://github.com/galvani/mautic/blob/feature/5.x-messenger/app/bundles/MessengerBundle/README.md)
2. open the landing page in anonymous window - note down current time
3. check the landing page's stats, now hit should be **NOT** recorded
4. wait like 2 minutes (for additional check)
4. run the messenger consumer, you may use verbose output:
   `php bin/console --env=dev -vvv messenger:consume --limit=1 hit`
5. the consumer should show somewhere a message is processed
    something like:
```
    16:48:39 INFO      [app] processed page hit #22
    [
      "hostname" => "28f1eeabdcc6",
      "pid" => 785
    ]
    16:48:39 INFO      [messenger] Message Mautic\MessengerBundle\Message\PageHitNotification handled by Mautic\MessengerBundle\MessageHandler\PageHitNotificationHandler::__invoke
    [
      "class" => "Mautic\MessengerBundle\Message\PageHitNotification",
      "handler" => "Mautic\MessengerBundle\MessageHandler\PageHitNotificationHandler::__invoke"
    ]
    [
      "hostname" => "28f1eeabdcc6",
      "pid" => 785
    ]
    16:48:39 INFO      [messenger] Mautic\MessengerBundle\Message\PageHitNotification was handled successfully (acknowledging to transport).
    [
      "class" => "Mautic\MessengerBundle\Message\PageHitNotification"
    ]
    [
      "hostname" => "28f1eeabdcc6",
      "pid" => 785
    ]   
```
6. Verify it is recorded on landing page

more fun:
7. Verify message is lovely JSON in rabbitmq admin

## Verify email tracking pixel
Almost same as above just email your self, copy the tracking image url and use as you use the landing page's url above. Mailhog does the job.


## Additional check
 * correct time is recorded, make a delay before hiting the page and running the consumer and verify the time matches the hit not the run of the consumer
